frontend/src/components/UploadedObjectInformation/index.tsx

```diff
@@ -0,0 +1,178 @@
+import { ArrowLeft, Loader } from "lucide-react";
+import {
+  OwnerRole,
+  UploadedObjectMetadata,
+} from "../../models/UploadedObjectMetadata";
+import { getTypeFromMetadata, handleFileDownload } from "../../utils/file";
+import { useUserStore } from "../../states/user";
+import { useCallback, useMemo, useState } from "react";
+import { Button } from "../common/Button";
+import { ApiService } from "../../services/api";
+import { ObjectShareModal } from "../Files/ObjectShareModal";
+import { ObjectDeleteModal } from "../Files/ObjectDeleteModal";
+
+export const UploadedObjectInformation = ({
+  object,
+}: {
+  object: UploadedObjectMetadata | null;
+}) => {
+  const [shareModalCid, setShareModalCid] = useState<string | null>(null);
+  const [deleteModalCid, setDeleteModalCid] = useState<string | null>(null);
+  const { user } = useUserStore();
```

Comment:
```suggestion
  const user = useUserStore((state) => state.user);
```
same reason as above